src/Diagnostics.RuntimeHost/Services/SourceWatcher/Watchers/GitHubWatcher.cs

```diff
@@ -48,6 +49,37 @@ public override void Start()
             StartPollingForChanges();
         }
 
+        public override async Task<Tuple<bool, Exception>> CreateOrUpdateDetector(DetectorPackage pkg)
+        {
+            if (pkg == null)
+            {
+                return new Tuple<bool, Exception>(false, new ArgumentNullException("pkg"));
+            }
+
+            Tuple<bool, Exception> output = new Tuple<bool, Exception>(true, null);
+
+            try
+            {
+                string detectorFilePath = $"{pkg.Id.ToLower()}/{pkg.Id.ToLower()}";
+
+                string csxFilePath = $"{detectorFilePath}.csx";
+                string dllFilePath = $"{detectorFilePath}.dll";
```

Comment:
Since we have this information right now. Maybe we can leverage this to build change history for each detector?

Reply:
Exactly. We should be able to do it right now also.